File: src/rails/game/special/ExchangeForShare.java
package rails.game.special;

import rails.common.LocalText;
import rails.common.parser.ConfigurationException;
import rails.common.parser.Tag;
import rails.game.*;
import rails.game.state.Item;
import rails.util.*;

public final class ExchangeForShare extends SpecialProperty {

    /** The public company of which a share can be obtained. */
    String publicCompanyName;

    /** The share size */
    int share;


    /**
     * Used by Configure (via reflection) only
     */
    public ExchangeForShare(Item parent, String id) {
        super(parent, id);
    }

    @Override
    public void configureFromXML(Tag tag) throws ConfigurationException {

        super.configureFromXML(tag);

        Tag swapTag = tag.getChild("ExchangeForShare");
        if (swapTag == null) {
            throw new ConfigurationException("<ExchangeForShare> tag missing");
        }

        publicCompanyName = swapTag.getAttributeAsString("company");
        if (!Util.hasValue(publicCompanyName))
            throw new ConfigurationException(
                    "ExchangeForShare: company name missing");
        share = swapTag.getAttributeAsInteger("share", 10);
    }

    public boolean isExecutionable() {
        // FIXME: Check if this works correctly
        // IT is better to rewrite this check
        return ((PrivateCompany)originalCompany).getOwner() instanceof Player;
    }

    /**
     * @return Returns the publicCompanyName.
     */
    public String getPublicCompanyName() {
        return publicCompanyName;
    }

    /**
     * @return Returns the share.
     */
    public int getShare() {
        return share;
    }

    public String getId() {
        return toString();
    }

    // FIXME: The toString() methods are removed, change calls to those
//    @Override
//    public String toString() {
//        return "Swap " + originalCompany.getId() + " for " + share
//               + "% share of " + publicCompanyName;
//    }

    @Override
    public String toMenu() {
        return LocalText.getText("SwapPrivateForCertificate",
                originalCompany.getId(),
                share,
                publicCompanyName );
    }
    
    public String getInfo() {
        return toMenu();
    }
}

File: dotnet/RailsLib.Net/Net/Game/Special/ExchangeForShare.cs
using GameLib.Net.Common;
using GameLib.Net.Common.Parser;
using System;
using System.Collections.Generic;
using System.Text;

namespace GameLib.Net.Game.Special
{
    public sealed class ExchangeForShare : SpecialProperty
    {
        /** The public company of which a share can be obtained. */
        string publicCompanyName;

        /** The share size */
        int share;


        /**
         * Used by Configure (via reflection) only
         */
        public ExchangeForShare(IRailsItem parent, string id) : base(parent, id)
        {
        }

        
    override public void ConfigureFromXML(Tag tag)
        {
            base.ConfigureFromXML(tag);

        Tag swapTag = tag.GetChild("ExchangeForShare");
        if (swapTag == null)
            {
            throw new ConfigurationException("<ExchangeForShare> tag missing");
    }

    publicCompanyName = swapTag.GetAttributeAsString("company");
            if (string.IsNullOrEmpty(publicCompanyName))
            {
                throw new ConfigurationException(
                        "ExchangeForShare: company name missing");
            }
    share = swapTag.GetAttributeAsInteger("share", 10);
    }

override public bool IsExecutionable
{
            get
            {
                // FIXME: Check if this works correctly
                // IT is better to rewrite this check
                return ((PrivateCompany)originalCompany).Owner is Player;
            }
}

/**
 * @return Returns the publicCompanyName.
 */
public string PublicCompanyName
{
            get
            {
                return publicCompanyName;
            }
}

/**
 * @return Returns the share.
 */
public int Share
{
            get
            {
                return share;
            }
}

    override public string ToText()
{
    return "Swap " + originalCompany.Id + " for " + share
           + "% share of " + publicCompanyName;
}

    override public string ToMenu()
{
    return LocalText.GetText("SwapPrivateForCertificate",
            originalCompany.Id,
            share,
            publicCompanyName);
}

override public string GetInfo()
{
    return ToMenu();
}
    }
}
